File: cpp/src/utils/branding.cpp
#include "horus/utils/branding.hpp"
#include <iostream>

namespace horus {
namespace utils {

void show_ascii_art() {
    const char* c_h = "\033[96m";
    const char* c_o = "\033[94m";
    const char* c_r = "\033[95m";
    const char* c_u = "\033[35m";
    const char* c_s = "\033[91m";
    const char* c_dim = "\033[2m";
    const char* c_reset = "\033[0m";

    std::cout << std::endl;

    std::cout << "    " << c_h << "\u2588\u2588\u2557  \u2588\u2588\u2557 " << c_reset
              << c_o << "\u2588\u2588\u2588\u2588\u2588\u2588\u2557 " << c_reset
              << c_r << "\u2588\u2588\u2588\u2588\u2588\u2588\u2557 " << c_reset
              << c_u << "\u2588\u2588\u2557   \u2588\u2588\u2557" << c_reset
              << c_s << "\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2557" << c_reset << std::endl;

    std::cout << "    " << c_h << "\u2588\u2588\u2551  \u2588\u2588\u2551" << c_reset
              << c_o << "\u2588\u2588\u2554\u2550\u2550\u2550\u2588\u2588\u2557" << c_reset
              << c_r << "\u2588\u2588\u2554\u2550\u2550\u2588\u2588\u2557" << c_reset
              << c_u << "\u2588\u2588\u2551   \u2588\u2588\u2551" << c_reset
              << c_s << "\u2588\u2588\u2554\u2550\u2550\u2550\u2550\u255d" << c_reset << std::endl;

    std::cout << "    " << c_h << "\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2551" << c_reset
              << c_o << "\u2588\u2588\u2551   \u2588\u2588\u2551" << c_reset
              << c_r << "\u2588\u2588\u2588\u2588\u2588\u2588\u2554\u255d" << c_reset
              << c_u << "\u2588\u2588\u2551   \u2588\u2588\u2551" << c_reset
              << c_s << "\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2557" << c_reset << std::endl;

    std::cout << "    " << c_h << "\u2588\u2588\u2554\u2550\u2550\u2588\u2588\u2551" << c_reset
              << c_o << "\u2588\u2588\u2551   \u2588\u2588\u2551" << c_reset
              << c_r << "\u2588\u2588\u2554\u2550\u2550\u2588\u2588\u2557" << c_reset
              << c_u << "\u2588\u2588\u2551   \u2588\u2588\u2551" << c_reset
              << c_s << "\u255a\u2550\u2550\u2550\u2550\u2588\u2588\u2551" << c_reset << std::endl;

    std::cout << "    " << c_h << "\u2588\u2588\u2551  \u2588\u2588\u2551" << c_reset
              << c_o << "\u255a\u2588\u2588\u2588\u2588\u2588\u2588\u2554\u255d" << c_reset
              << c_r << "\u2588\u2588\u2551  \u2588\u2588\u2551" << c_reset
              << c_u << "\u255a\u2588\u2588\u2588\u2588\u2588\u2588\u2554\u255d" << c_reset
              << c_s << "\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2551" << c_reset << std::endl;

    std::cout << "    " << c_h << "\u255a\u2550\u255d  \u255a\u2550\u255d " << c_reset
              << c_o << "\u255a\u2550\u2550\u2550\u2550\u2550\u255d " << c_reset
              << c_r << "\u255a\u2550\u255d  \u255a\u2550\u255d " << c_reset
              << c_u << "\u255a\u2550\u2550\u2550\u2550\u2550\u255d " << c_reset
              << c_s << "\u255a\u2550\u2550\u2550\u2550\u2550\u2550\u255d" << c_reset << std::endl;

    std::cout << std::endl;
    std::cout << "    " << c_o << "        Holistic Operational Reality" << c_reset << std::endl;
    std::cout << "    " << c_r << "            for Unified Systems" << c_reset << std::endl;
    std::cout << std::endl;
    std::cout << "    " << c_dim << "\xF0\x9F\xA4\x96 Mixed Reality Robot Management Platform" << c_reset << std::endl;
    std::cout << "    " << c_dim << "\xF0\x9F\x8F\x97\xEF\xB8\x8F Developed at RICE Lab, University of Genoa" << c_reset << std::endl;
    std::cout << "    " << c_dim << "HORUS SDK v" << VERSION << c_reset << std::endl;
    std::cout << std::endl;

    std::cout << "============================================================" << std::endl;
}

} // namespace utils
} // namespace horus

File: rust/src/utils/branding.rs
/// HORUS SDK version
pub const VERSION: &str = env!("CARGO_PKG_VERSION");

/// Display HORUS ASCII art header with version information
pub fn show_ascii_art() {
    let c_h = "\x1b[96m";
    let c_o = "\x1b[94m";
    let c_r = "\x1b[95m";
    let c_u = "\x1b[35m";
    let c_s = "\x1b[91m";
    let c_dim = "\x1b[2m";
    let c_reset = "\x1b[0m";

    let rows = [
        (
            "\u{2588}\u{2588}\u{2557}  \u{2588}\u{2588}\u{2557} ",
            "\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2557} ",
            "\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2557} ",
            "\u{2588}\u{2588}\u{2557}   \u{2588}\u{2588}\u{2557}",
            "\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2557}",
        ),
        (
            "\u{2588}\u{2588}\u{2551}  \u{2588}\u{2588}\u{2551}",
            "\u{2588}\u{2588}\u{2554}\u{2550}\u{2550}\u{2550}\u{2588}\u{2588}\u{2557}",
            "\u{2588}\u{2588}\u{2554}\u{2550}\u{2550}\u{2588}\u{2588}\u{2557}",
            "\u{2588}\u{2588}\u{2551}   \u{2588}\u{2588}\u{2551}",
            "\u{2588}\u{2588}\u{2554}\u{2550}\u{2550}\u{2550}\u{2550}\u{255d}",
        ),
        (
            "\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2551}",
            "\u{2588}\u{2588}\u{2551}   \u{2588}\u{2588}\u{2551}",
            "\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2554}\u{255d}",
            "\u{2588}\u{2588}\u{2551}   \u{2588}\u{2588}\u{2551}",
            "\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2557}",
        ),
        (
            "\u{2588}\u{2588}\u{2554}\u{2550}\u{2550}\u{2588}\u{2588}\u{2551}",
            "\u{2588}\u{2588}\u{2551}   \u{2588}\u{2588}\u{2551}",
            "\u{2588}\u{2588}\u{2554}\u{2550}\u{2550}\u{2588}\u{2588}\u{2557}",
            "\u{2588}\u{2588}\u{2551}   \u{2588}\u{2588}\u{2551}",
            "\u{255a}\u{2550}\u{2550}\u{2550}\u{2550}\u{2588}\u{2588}\u{2551}",
        ),
        (
            "\u{2588}\u{2588}\u{2551}  \u{2588}\u{2588}\u{2551}",
            "\u{255a}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2554}\u{255d}",
            "\u{2588}\u{2588}\u{2551}  \u{2588}\u{2588}\u{2551}",
            "\u{255a}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2554}\u{255d}",
            "\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2588}\u{2551}",
        ),
        (
            "\u{255a}\u{2550}\u{255d}  \u{255a}\u{2550}\u{255d} ",
            "\u{255a}\u{2550}\u{2550}\u{2550}\u{2550}\u{2550}\u{255d} ",
            "\u{255a}\u{2550}\u{255d}  \u{255a}\u{2550}\u{255d} ",
            "\u{255a}\u{2550}\u{2550}\u{2550}\u{2550}\u{2550}\u{255d} ",
            "\u{255a}\u{2550}\u{2550}\u{2550}\u{2550}\u{2550}\u{2550}\u{255d}",
        ),
    ];

    println!();
    for (h, o, r, u, s) in rows {
        println!(
            "    {}{}{}{}{}{}{}{}{}{}{}{}{}{}{}",
            c_h, h, c_reset,
            c_o, o, c_reset,
            c_r, r, c_reset,
            c_u, u, c_reset,
            c_s, s, c_reset,
        );
    }

    println!();
    println!("    {}        Holistic Operational Reality{}", c_o, c_reset);
    println!("    {}            for Unified Systems{}", c_r, c_reset);
    println!();
    println!("    {}\u{1F916} Mixed Reality Robot Management Platform{}", c_dim, c_reset);
    println!(
        "    {}\u{1F3D7}\u{FE0F} Developed at RICE Lab, University of Genoa{}",
        c_dim, c_reset
    );
    println!("    {}HORUS SDK v{}{}", c_dim, VERSION, c_reset);
    println!();

    println!("============================================================");
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_version_not_empty() {
        assert!(!VERSION.is_empty());
    }

    #[test]
    fn test_show_ascii_art() {
        // Just make sure it doesn't panic
        show_ascii_art();
    }
}
